metricbeat/module/docker/memory/data.go

```diff
@@ -29,20 +29,31 @@ func eventsMapping(r mb.ReporterV2, memoryDataList []MemoryData) {
 }
 
 func eventMapping(r mb.ReporterV2, memoryData *MemoryData) {
-	fields := common.MapStr{
-		"fail": common.MapStr{
-			"count": memoryData.Failcnt,
-		},
-		"limit": memoryData.Limit,
-		"rss": common.MapStr{
-			"total": memoryData.TotalRss,
-			"pct":   memoryData.TotalRssP,
-		},
-		"usage": common.MapStr{
-			"total": memoryData.Usage,
-			"pct":   memoryData.UsageP,
-			"max":   memoryData.MaxUsage,
-		},
+
+	//if we have windows memory data, just report windows stats
+	var fields common.MapStr
+	if memoryData.Commit+memoryData.CommitPeak+memoryData.PrivateWorkingSet > 0 {
+		fields = common.MapStr{
+			"commit":              memoryData.Commit,
+			"commit_peak":         memoryData.CommitPeak,
+			"private_working_set": memoryData.PrivateWorkingSet,
+		}
+	} else {
+		fields = common.MapStr{
+			"fail": common.MapStr{
+				"count": memoryData.Failcnt,
+			},
+			"limit": memoryData.Limit,
+			"rss": common.MapStr{
+				"total": memoryData.TotalRss,
+				"pct":   memoryData.TotalRssP,
+			},
+			"usage": common.MapStr{
+				"total": memoryData.Usage,
+				"pct":   memoryData.UsageP,
+				"max":   memoryData.MaxUsage,
+			},
+		}
```

Comment:
I am fine with this approach if we cannot consider these metrics equivalent to others on Linux. The private working set could be similar to the RSS, and commit and commit peak could be similar to usage and max usage, but not sure.
Probably @narph or @andrewkroh can shed more light about this.
If they are equivalent or similar enough we should send them in the same fields.

Reply:
Commit - memory space the application has reserved for use, so this will be allocated but not necessarily used. (This is composed of main memory (RAM) and disk (pagefiles).)
CommitPeak - the highest amount that the commit has reached since the operating system was last started or session.
PrivateWorkingSet - the amount of memory used by a process.
Not sure the memory stats are matching here so I would go for separate naming, we might want to look into that as well for the other memory related metricsets